Add support for workspace projects

slheavner opened the pull request:
`npm ls` for workspace projects include the "root" project, resulting in ropm including the workspace project as a dependency of itself, and adding postfixes (_v1, etc.) to its actual dependencies

How:
- use `--json` for `npm ls` instead of `--parseable`
- iterate items in the json dependency try until we find the matching `name` field
- use that item as the "root" dependency graph
- flatten the json to match the original file path format

TwitchBronBron commented:
Looks like it's still failing.

TwitchBronBron commented:
oh wow, I completely forgot about this! I'll try and give it a good review this week. Sorry about that!

slheavner commented:
no worries, revisiting. 

There the problem with CI is the npm version tagged there vs what I was testing against. I need to dig into the differences and find what's causing it.

Commit:
Add suppoort for npm workspaces

- `npm ls` always lists dependencies from the root project, even if
  running from a workspace project. This changes uses the `--json` and
`--long` options in order to only parse dependencies from the current
project tree
- note: node_modules loctiaon changes significantly between npm versions

slheavner commented:
workflow passed on my fork, should be good to go

TwitchBronBron commented:
I'm so sorry this fell through the cracks for so long. Do you mind taking a look at the merge conflicts and letting us know if this is still something that is needed?

Also, I'm not entirely familiar with "workspace projects", so do you mind giving a quick overview of that that actually means? Once I have a better understanding of what this solves, I'm happy to merge it and cut a release.

addison-adler commented:
@slheavner not sure if you're still using (or trying to use) npm workspaces with ropm, but wanted to let you know:

1. I'm also using npm workspaces and am working on updates to `ropm` this week to support workspaces.
2. as a workaround, if you include the workspace dependency as a listed dependency with a version wildcard, it will pick up the right name. Example:

```json
"workspaces": [
    "./src-roku/roku-log"
],
"dependencies": {
    "roku-log": "*"
}
```

TwitchBronBron committed:
Merge branch 'master' of https://github.com/rokucommunity/ropm into feature/workspace-support

TwitchBronBron committed:
Fix tests for multiple node versions

TwitchBronBron committed:
Increase buffer size for projects with many dependencies

TwitchBronBron reviewed (approved):
Sorry this took so long to lang. I finally ran into a situation where I needed this as well, so was able to reproduce it and test it. Looks great!

I made some additional changes:
- support running the test on any node version, it auto-adjusts to the right expected test output. 
- ensure that pnpm installed projects also don't crash the system by increasing the buffer (since pnpm is a popular way of doing workspaces)
